packages/@sanity/components/src/previews/BlockImagePreview.js

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import styles from 'part:@sanity/components/previews/block-image-style'
+
+const fieldProp = PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func])
+
+export default class BlockImagePreview extends React.PureComponent {
+  static propTypes = {
+    title: PropTypes.string,
+    subtitle: PropTypes.string,
+    description: PropTypes.string,
+    mediaDimensions: PropTypes.shape({
+      width: PropTypes.number,
+      height: PropTypes.number,
+      fit: PropTypes.oneOf(['clip', 'crop', 'clamp']),
+      aspect: PropTypes.number,
+    }),
+    media: fieldProp,
+    children: PropTypes.func,
+    type: PropTypes.shape({
+      title: PropTypes.string,
+      name: PropTypes.string
+    })
+  }
+
+  static defaultProps = {
+    mediaDimensions: {width: 300, height: 200, aspect: 1, fit: 'crop'},
+    type: {
+      title: undefined,
+      name: undefined
+    }
+  }
+
+  render() {
+    const {
+      title,
+      subtitle,
+      description,
+      mediaDimensions,
+      media,
+      children
+    } = this.props
+
+    return (
+      <div
+        className={styles.root}
+      >
+        {
+          media && (
+            <div className={`${styles.media}`}>
+              {
+                typeof media === 'function' && (
+                  media({dimensions: mediaDimensions, layout: 'blockImage'})
+                )
+              }
+              {
+                typeof media === 'string' && (
+                  <div className={styles.mediaString}>{media}</div>
+                )
+              }
+              {
+                typeof media === 'object' && media
```

Comment:
Not clear to me what this does?

Reply:
We use this pattern in the other previews. I remember discussing this, and concluding that an element is an object. In the proptypes we accept `PropTypes.string, PropTypes.node, PropTypes.func`. We handle `string` and `function` here, but a node will be an `object`. If a JSON-object or other object is provided, the proptypes will complain.

Reply:
Ah, right, I forgot that a React elements has typeof object. You should probably use `React.isValidElement(media)` instead to avoid confusion. Regular objects are *not* valid as a React element and will throw an error, so this line sure looks like a bug to the reader :)

Reply:
👍